webapp/src/components/Staking/StakingOverview.tsx

```diff
@@ -115,6 +115,8 @@ const OverviewLabel = styled(SecondaryText)`
 const StakingOverview = () => {
   const { stakingPools, loading: stakingLoading } = useStakingPool(VaultList);
   const { data: tokenData, loading: tokenLoading } = useRBNToken();
+  const [week, setWeek] = useState(1);
+  const [nextStakeRewardStart, setNextStakeRewardStart] = useState(moment.utc("2021-06-18").set("hour", 10).set("minute", 30));
```

Comment:
In general, we should avoid state whenever we can compute the information with existing data. One way to rewrite this is to remove these two lines and do this instead:

```js
const [week, nextStakeRewardStart] = useMemo(() => {

  return [week, nextStakeRewardStart]

}, [week, nextStakeRewardStart]);
```

Reply:
Something like this?
```javascript
  let [week, nextStakeRewardStart] = useMemo(() => {
    return [1, moment.utc("2021-06-18").set("hour", 10).set("minute", 30)]
  }, []);
```
I still dont really understand why you cant use state

Reply:
I think there is several reason why using state might not be ideal over here. 

The first one being that the date for nextStakeRewardStart are pretty much derivation of current date + calculation, so we can avoid `setNextStakeRewardStart` at all. Besides, minimize the number of state when not necessary helps with readability in long run as well.  

I think a better example would be 

```
  const [week, nextStakeRewardStart] = useMemo(() => {
    const startDate = moment.utc("2021-06-18").set("hour", 10).set("minute", 30);

    // Derive week count directly
    // +1 because asWeeks return full week, as we know, first week is week one, where as when a full week past, it's 1 week + 1
    const weekCount = moment.duration(startDate.diff(moment())).asWeeks() + 1

    // Derive next stake reward start here
   const nextStakeRewardStart = .....

    return [weekCount, nextStakeRewardStart]
  }, []);
```

Reply:
Ive made the following changes. Would this mean that weekCount/week wont rerender when it changes?
```javascript
  const [week, nextStakeRewardStart] = useMemo(() => {
    const startDate = moment.utc("2021-06-18").set("hour", 10).set("minute", 30);

    let weekCount
  
    if(moment().diff(startDate) < 0){
      weekCount = 1
    } else {
      weekCount = moment().diff(startDate, "weeks") + 2
    }
    
    // Next stake reward date
    const nextStakeReward = startDate.add(7 * (weekCount - 1), "days")

    return [weekCount, nextStakeReward]
  }, []);

  const timeTillNextRewardWeek = useMemo(() => {
    const endStakeReward = moment.utc("2021-06-18").set("hour", 10).set("minute", 30).add("28", "days");

    if (endStakeReward.diff(moment()) <= 0) {
      return "End of Rewards, you can unstake now";
    }

    // Time till next stake reward date
    const startTime = moment.duration(
      nextStakeRewardStart.diff(moment()),
      "milliseconds"
    );

    return `${startTime.days()}D ${startTime.hours()}H ${startTime.minutes()}M`;
  }, [nextStakeRewardStart]);
  ```

Reply:
That looks better. Yep, the users can just refresh if they want to see it countdown